packages/authenticator/amplify_authenticator/lib/src/utils/validators.dart

```diff
@@ -28,6 +33,32 @@ FormFieldValidator<String> simpleValidator(
   };
 }
 
+FormFieldValidator<String> usernameValidator({
+  required bool isOptional,
+  required BuildContext context,
+  required InputResolver inputResolver,
+}) {
+  return (String? input) {
+    if (isOptional) return null;
```

Comment:
Should we still validate the input if it's provided?

Reply:
Username cannot be optional, so this line should never actually be hit. The code is confusing though. We can remove  `isOptional` as it is always false in this case.

Thanks for calling that out. I pushed an update.